core/src/main/scala/org/http4s/headers/Forwarded.scala

```diff
@@ -121,7 +168,34 @@ object Forwarded
 
     /** Parses host and optional port number from the given string according to RFC3986.
       */
-    def fromString(s: String): ParseResult[Host] = new ModelHostParser(s).parse
+    def fromString(s: String): ParseResult[Host] =
+      parser.parseAll(s).left.map { e =>
+        ParseFailure(s"invalid host '$s'", e.toString)
+      }
+
+    val parser: Parser0[Host] = {
+      // this is awkward but the spec allows an empty port number
+      val port: P[Option[Int]] = Numbers.digits
+        .mapFilter { s =>
+          if (s.isEmpty) Some(None)
+          else {
+            try {
+              val i = s.toInt
+              Option.when(i <= PortMax)(Some(i))
```

Comment:
`Option.when` is a 2.13+ api, so that needs to be replaced with `if (i <= PortMax) Some(Some(i)) else None`